### PR TITLE
.NET update to 0.12.0-beta & Emojification of manual spans

### PR DESCRIPTION
**Pull request by robbkidd**

### CODE REVIEW

File: dotnet/message-service/Controllers/MessageController.cs [hunk 1]
```diff
@@ -24,6 +24,8 @@ public async Task<string> GetAsync()
         {
 
             var message = await DetermineMessage();
+            var currentSpan = Tracer.CurrentSpan;
+            currentSpan.SetAttribute("app.message", message);
```
Comment (robbkidd) [inline dotnet/message-service/Controllers/MessageController.cs:28]: We figured out how to get the current span!

File: dotnet/year-service/Controllers/YearController.cs [hunk 2]
```diff
@@ -16,25 +16,17 @@ public class YearController : ControllerBase
...
-            using var activity = Startup.ActivitySource.StartActivity("DetermineYear");
+            using var activity = Startup.ActivitySource.StartActivity("🗓 get-a-year ✨");
             activity?.SetTag("banana", 1);
             var year = await DetermineYear();
             return year;
         }
 
         private static async Task<int> DetermineYear()
         {
-            await SleepAwhile();
```
Comment (robbkidd) [inline dotnet/year-service/Controllers/YearController.cs:27]: Opted not to make the spans longer than actual code takes to run.

Comment (JamieDanielson) [reply dotnet/year-service/Controllers/YearController.cs:27]: Understandable, though quick note: This is now an async method with no await.

**Review by JamieDanielson — APPROVED** [review]
We'll want to re-review that async DetermineYear at some point but it's a warning, not an error, and code checks out. Also consider adding some additional notes around adding OTLP endpoint info to env var.

File: dotnet/year-service/Controllers/YearController.cs (hunk 2 above)
Comment (MikeGoldsmith) [reply dotnet/year-service/Controllers/YearController.cs:27]: Yep, agree with @JamieDanielson - without an await, you can remove the async method modifier 👍🏻

File: dotnet/frontend/Controllers/GreetingController.cs [hunk 3]
```diff
@@ -48,27 +48,31 @@ public async Task<string> GetAsync()
...
-
                 span.SetAttribute("testAttribute", "Greeting");
                 Baggage.Current.SetBaggage("testBaggage", "Greetings");
                 var httpClient = _clientFactory.CreateClient();
                 var name = await GetNameAsync(httpClient);
-                var message = await GetMessage(httpClient);
+                var message = await GetMessageAsync(httpClient);
 
-                return $"Hello {name}, {message}";
+                using var render_span = _tracer.StartActiveSpan("🎨 render greeting ✨");
```
Comment (MikeGoldsmith) [inline dotnet/frontend/Controllers/GreetingController.cs:57]: .NET prefers camelCase for local variables.

```suggestion
                using var renderSpan = _tracer.StartActiveSpan("🎨 render greeting ✨");
```

**Review by MikeGoldsmith — APPROVED** [review]
Looks good - just a couple of minor changes 👍🏻

File: dotnet/frontend/Controllers/GreetingController.cs (hunk 3 above)
Comment (robbkidd) [reply dotnet/frontend/Controllers/GreetingController.cs:57]: Aaaa! My Ruby's showing.

File: dotnet/year-service/Controllers/YearController.cs (hunk 2 above)
Comment (robbkidd) [reply dotnet/year-service/Controllers/YearController.cs:27]: Removed all the async from Year since there is nothing to await.

**Pull request merged**